pyhealth/calib/utils.py

```diff
@@ -0,0 +1,71 @@
+from collections import defaultdict
+from typing import Optional
+
+import numpy as np
+import torch
+import tqdm
+from torch import Tensor
+
+from pyhealth.datasets import utils as datautils
+
+
+def agg_loss(loss:torch.Tensor, reduction: str):
+    if reduction == 'mean':
+        return loss.mean()
+    if reduction == 'sum':
+        return loss.sum()
+    return loss
+
+
+def one_hot_np(labels, K):
+    new_labels = np.zeros((len(labels), K))
+    new_labels[np.arange(len(labels)), labels] = 1
+    return new_labels
+
+class LogLoss(torch.nn.Module):
+    """Cross entropy, but takes in the probability instead of the logits"""
+    reduction: str
+    def __init__(self, weight: Optional[Tensor] = None,  ignore_index: int = -100, reduction: str = 'mean', clip=1e-10) -> None:
+        super(LogLoss, self).__init__()
+        self.register_buffer('weight', weight)
+        self.ignore_index = ignore_index
+        self.reduction = reduction
+        self.clip = clip
```

Comment:
what is Line27 used for?

Reply:
reduction is just specifying whether we should reduce the loss by mean/sum etc.

I was trying to follow https://pytorch.org/docs/1.10/_modules/torch/nn/modules/loss.html#CrossEntropyLoss in how they write loss

Reply:
ok, I thought you already define "reduction: str" in the function argument, and it would be unnecessary. But never mind.